docs/writing_your_plugin.md

```diff
@@ -77,41 +72,66 @@ functionality this way.
             virtual void start() override { }
             virtual ~plugin_name() override { }
         };
-
+        
         // This line makes the plugin importable by Spindle
         PLUGIN_MAIN(plugin_name);
 
 
-4.  At this point, you should be able to go to the ILLIXR root and `make dbg`. If you edit a source
-    file and then `make dbg`, it should trigger a rebuild of your plugin.
-
-5.  This is all that is required to be a plugin which can be loaded by Spindle in the ILLIXR
-    runtime. Reading and writing from Phonebook and Switchboard optional, but nearly every plugin
-    does it. See `default_plugins.md` for more details.
-
-    First, we can query the [`phonebook`][2] to get various services including [`switchboard`][5]. Then we
-    query [`switchboard`][5] for event-streams (topics). We will read `topic1`, write to `topic2`, and
-    schedule computation on `topic 3`. See the API documentation for `phonebook` and `switchboard`
-    for more details.
+4. At this point, you should be able to build your plugin with ILLIXR. Move to the ILLIXR repo and update `configs/native.yaml`. See [Building ILLIXR][1] for more details on the config file.
```

Comment:
Need some info on what to do when the new plugin replaces an old one. For example, can't have two SLAMs running at the same time and publishing to the same topic.